src/Util.hpp

```diff
@@ -33,6 +35,22 @@
 
 class Context;
 
+// Helper variadic template function to construct native paths.
+// Used like: PATH("usr", "local", "bin");
+template<typename... T>
+std::string
+PATH(T&... args)
+{
+  std::initializer_list<const std::string> paths{args...};
```

Comment:
Since we support C++17 because it's where `<filesystem>` is supported there's actually a way to implement the whole function basically in one line - `return (std::filesystem::path {} / ... / args).string ()`

Reply:
That's amazing. Can you please explain how this works or give me some references to read?

Reply:
Here's a reference:
https://en.cppreference.com/w/cpp/language/fold